tornettools/generate_tgen.py

```diff
@@ -10,22 +12,33 @@
 from tornettools.generate_defaults import *
 from tornettools.util import load_json_data
 
+from Crypto.PublicKey import RSA
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.backends import default_backend
+
 def generate_tgen_config(args, tgen_clients, tgen_servers):
     # make sure the config directory exists
     abs_conf_path = "{}/{}".format(args.prefix, CONFIG_DIRPATH)
     if not os.path.exists(abs_conf_path):
         os.makedirs(abs_conf_path)
 
-    server_peers = ["{}:{}".format(server['name'], TGEN_SERVER_PORT) for server in tgen_servers]
+    if args.hidden:
+        server_peers = ["{}:{}".format(server['hs_onion_url'], TGEN_SERVER_PORT) for server in tgen_servers]
+    else:
+        server_peers = ["{}:{}".format(server['name'], TGEN_SERVER_PORT) for server in tgen_servers]
 
-    __generate_tgenrc_server(abs_conf_path)
+    __generate_tgenrc_server(args, abs_conf_path)
     __generate_tgenrc_perfclient(abs_conf_path, server_peers)
     __generate_tgenrc_markovclients(abs_conf_path, tgen_clients, server_peers)
     __generate_tgen_traffic_models(args, abs_conf_path)
 
-def __generate_tgenrc_server(abs_conf_path):
+def __generate_tgenrc_server(args, abs_conf_path):
     G = DiGraph()
-    G.add_node("start", serverport="{}".format(TGEN_SERVER_PORT), loglevel="message", stallout="0 seconds", timeout="0 seconds")
+    if args.hidden:
+        G.add_node("start", serverport="{}".format(TGEN_HIDDENSERVICE_PORT), loglevel="message", stallout="0 seconds", timeout="0 seconds")
+    else:
+        G.add_node("start", serverport="{}".format(TGEN_SERVER_PORT), loglevel="message", stallout="0 seconds", timeout="0 seconds")
```

Comment:
How about the tgen server process always listens on `TGEN_SERVER_PORT` (80)?

(and then for servers that are also available as onion services, the onion service listens on `TGEN_HIDDENSERVICE_PORT` and redirects to `TGEN_SERVER_PORT`)

Reply:
Done.